controllers/datamovement_controller.go

```diff
@@ -89,314 +90,275 @@ func (r *DataMovementReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
-			msg := fmt.Sprintf("Failed to monitor: %v", err)
-			resetConditionRollup(dm, nnfv1alpha1.DataMovementConditionReasonFailed, msg)
+		// TODO: Capture output as it progresses and add a %complete to the resource
 
-			if err := r.Status().Update(ctx, dm); err != nil {
-				log.Error(err, "Failed to record monitor error status")
-				return ctrl.Result{}, err
-			}
+		out, err := cmd.CombinedOutput()
 
-			return ctrl.Result{}, err
-		} else if result != nil && !result.IsZero() {
-			return *result, nil
+		now := metav1.NowMicro()
+		dm.Status.EndTime = &now
+		dm.Status.State = nnfv1alpha1.DataMovementConditionTypeFinished
+		dm.Status.Status = nnfv1alpha1.DataMovementConditionReasonSuccess
+
+		if errors.Is(ctxCancel.Err(), context.Canceled) {
+			log.Error(err, "Data movement operation cancelled", "output", string(out))
+			dm.Status.Status = nnfv1alpha1.DataMovementConditionReasonCancelled
+		} else if err != nil {
+			log.Error(err, "Data movement operation failed", "output", string(out))
+			dm.Status.Status = nnfv1alpha1.DataMovementConditionReasonFailed
+			dm.Status.Message = err.Error()
+
+			// TODO: Enhanced error capture: parse error response and provide useful message
+		} else {
+			log.Info("Completed Command", "cmd", cmd.String())
 		}
 
-		// Continue monitoring the resource, if specified, by preventing the data movement resource from finishing
-		// early while the client application is running and generating data movement requests. This is to protect
-		// the use case where this resource thinks it handled all the requests but in reality the client application
-		// continues to execute with the potential to generate more requests.
-		if dm.Spec.Monitor == true {
-			return ctrl.Result{}, nil
-		}
+		status := dm.Status.DeepCopy()
 
-		switch status {
-		case nnfv1alpha1.DataMovementConditionTypeRunning:
-			// Still running, nothing to do here
-			break
-		case nnfv1alpha1.DataMovementConditionReasonFailed, nnfv1alpha1.DataMovementConditionReasonSuccess:
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			dm := &nnfv1alpha1.NnfDataMovement{}
+			if err := r.Get(ctx, req.NamespacedName, dm); err != nil {
+				return client.IgnoreNotFound(err)
+			}
 
-			advanceCondition(dm,
-				nnfv1alpha1.DataMovementConditionTypeFinished,
-				status,
-				message,
-			)
+			status.DeepCopyInto(&dm.Status)
 
-			now := metav1.NowMicro()
-			dm.Status.EndTime = &now
-		}
+			return r.Status().Update(ctx, dm)
+		})
 
-		if err := r.Status().Update(ctx, dm); err != nil {
-			log.Error(err, "Failed to transition to finished state")
-			return ctrl.Result{}, err
+		if err != nil {
+			log.Error(err, "failed to update rsync status with completion")
+			// TODO Add prometheus counter to track occurrences
 		}
 
-		return ctrl.Result{}, nil
-
-	case nnfv1alpha1.DataMovementConditionTypeFinished:
-		return ctrl.Result{}, nil // Already finished, do nothing further.
-	}
+		r.contexts.Delete(dm.Name)
```

Comment:
Shouldn't need this since you do a `LoadAndDelete()` above. But maybe we should just do a `Load()` above in case the cancel fails.